packages/repository/src/repositories/relation.repository.ts

```diff
@@ -144,3 +145,69 @@ export class DefaultHasManyEntityCrudRepositorys<T extends Entity, ID>
     throw new Error('Method not implemented.');
   }
 }
+
+export class RelationFactory<
+  Source extends Entity,
+  Target extends Entity,
+  TargetRepository extends DefaultCrudRepository<
```

Comment:
TargetRepository is only required to extend the `EntityCrudRepository` interface instead of a base class.